sql/core/src/main/scala/org/apache/spark/sql/internal/SharedState.scala

```diff
@@ -191,11 +194,15 @@ private[sql] class SharedState(
 }
 
 object SharedState extends Logging {
-  try {
-    URL.setURLStreamHandlerFactory(new FsUrlStreamHandlerFactory())
-  } catch {
-    case e: Error =>
-      logWarning("URL.setURLStreamHandlerFactory failed to set FsUrlStreamHandlerFactory")
+  private def setFsUrlStreamHandlerFactory(conf: SparkConf): Unit = {
+    if (conf.get(DEFAULT_URL_STREAM_HANDLER_FACTORY_ENABLED)) {
+      try {
+        URL.setURLStreamHandlerFactory(new FsUrlStreamHandlerFactory())
+      } catch {
+        case _: Error =>
```

Comment:
NonFatal? this won't catch exceptions I think. And do you want to log the exception?

Reply:
Got it. `SecurityException` can be thrown. I'll update it to `NonFatal`.